Automattic/woocommerce-payments
# Remove `simplifyDepositsUi` feature flag and legacy UI code

Fixes #5749

#### Changes proposed in this Pull Request

This PR removes the `simplifyDepositsUi` feature flag which is now the default UI, removing legacy UI code and improving code quality in the process.

* 🧨 Remove the `simplifyDepositsUi` feature flag which was previously enabled by default.
* 🧨 Remove the legacy `<DepositsInformation />` card component that has been superseded by the `<AccountBalances />` and `<DepositsOverview />` card components (`client/deposits/utils`). This removed card was not visible due to the `simplifyDepositsUi` feature flag being enabled by default.
* 🧨 Remove functions from `client/deposits/utils` that were only used by the legacy `<DepositsInformation />` component.
* 🧨 Remove the legacy variant of the `<InstantDepositButton />` button with the new variant introduced in #6004.
* 🔧 Fixes tests that were affected by the above changes.
* ♻️ Refactor the `client/deposits/utils` code to TypeScript to improve code quality.
* ♻️ Refactor the `<InstantDepositButton />` and `<InstantDepositModal />` component to TypeScript to improve code quality.


#### Testing instructions





* Visit the **Payments → Overview** screen
* Ensure the the Account Balances and Deposits cards render as expected.
* With a WCPay account that is eligible for Instant Deposits, ensure that the Instant Deposit button renders as expected and the modal appears on click.
  * ℹ️ I have a test site that is eligible for Instant Deposits accessible via _secret store_ with ID `10962` (search for `ej-wcpay-usa` on _secret store_). I've uploaded this PR's built WCPay plugin to this site to facilitate testing this PR.
* Ensure all tests and GH checks pass.

Here is a screenshot of the Account Balances and Deposits cards **before/after** this PR for an account eligible for instant deposits.

<div style="display: flex; flex-direction: row;">

<img width="49%" alt="Before (develop branch)" src="https://github.com/Automattic/woocommerce-payments/assets/3147296/acec9436-8a8c-46a6-9477-c9170b31b4e0" />

<img width="49%" alt="After (this PR)" src="https://github.com/Automattic/woocommerce-payments/assets/3147296/34519cb2-8318-4e2d-81bb-4e939da27d5f" />
</div>


-------------------


**Post merge**
